### PR TITLE
creating  radio and checkbox components in dp-renderer

### PR DESCRIPTION
**Pull request by yomiodeleye**
### What

creating radio and checkbox componets for globaly usage

### How to review

by bringing the partial template into your project

### Who can review

FD

### CODE REVIEW

**Review by gilesburdett — CHANGES_REQUESTED** [review]
Some observations common to both templates:

- As a template is returning two elements that are siblings, one shouldn't be indented more than the other
- Where an element has several attributes each attribute can go on its own line so we don't get long lines of attributes (easier to compare code side by side etc), and this has no impact on element layout when the HTML is parsed
- For localisation we recently adopted a new approach as standard in dp-renderer. Please see `LOCALISATION.md` in the root of the repository for more information and let me know if there's anything missing or unclear
- Each template will need a new model in `/model` associated with it
- The count in parentheses is something specific to the Release Calendar, and other users of this component will be using it in different circumstances that don't need this. In React this might have been a job for a higher order component but we don't have that luxury with Go Templates so another way will have to be found.

File: assets/templates/partials/label.tmpl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<label
+  class="ons-radio__label"
```
Comment (gilesburdett) [inline assets/templates/partials/label.tmpl:2]: A label for a radio or a checkbox have different classes:
- `ons-radio__label`
- `ons-checkbox__label`

File: assets/templates/partials/inputs/input-radio.tmpl [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  id="{{ .Id }}"
+  name="{{ .Name }}"
+  class="ons-radio__input ons-js-radio"
+  value="{{ .Value }}"
+  {{ if .Checked }}
+    checked
+  {{ end }}
+>  
+{{ if .LocaleKey }}
+  {{ template "partials/label" .}}
```
Comment (gilesburdett) [inline assets/templates/partials/inputs/input-radio.tmpl:12]: In the wider scheme of things in dp-renderer, calling this partial "label" could be misunderstood as a label to be associated for any kind of element, when it is in fact only to be used with radio or checkbox inputs.

It really needs a more specific name.

This also applies to the reference in `input-checkbox.tmpl`

**Review by gilesburdett — CHANGES_REQUESTED** [review]

File: assets/templates/partials/inputs/input-radio.tmpl (hunk 2 above)
Comment (yomiodeleye) [reply assets/templates/partials/inputs/input-radio.tmpl]: In this case i think is better to move the label into their different input and conditional wrap count. Regarding  naming. what do you mean?

File: assets/templates/partials/inputs/input-radio.tmpl [hunk 3]
```diff
@@ -9,5 +9,14 @@
   {{ end }}
 >  
 {{ if .LocaleKey }}
-  {{ template "partials/label" .}}
+  <label
+    class="ons-radio__label"
+    for="{{.Id}}"
+    id="{{.Id}}-label"
+    >
```
Comment (gilesburdett) [inline assets/templates/partials/inputs/input-radio.tmpl:16]: Line 16 - Match indentation of `>` with opening `<label`

**Review by gilesburdett — CHANGES_REQUESTED** [review]
The `{{ if .LocaleKey }}` around the `<label>` is redudant and can be removed.

File: assets/templates/partials/inputs/input-radio.tmpl (hunk 3 above)
Comment (yomiodeleye) [reply assets/templates/partials/inputs/input-radio.tmpl]: this is done

**Review by gilesburdett — APPROVED** [review]

**Pull request merged**